File: 007/Problem7.cs
using System;

namespace EulerProblems
{
    class Program
    {
        static void Main(string[] args)
        {
            var watch = new System.Diagnostics.Stopwatch();

            watch.Start();
            Problem();
            watch.Stop();

            Console.WriteLine($"Function Took: {watch.Elapsed}");
        }

        private static bool SlowPrime(int num)
        {
            for (var i = 2; i < num; i++)
            {
                if (num % i == 0)
                {
                    return false;
                }
            }
            return true;
        }
        private static void Problem()
        {
            var count = 2;
            var i = 4;
            while (true)
            {
                i++;
                if (SlowPrime(i))
                {
                    count++;
                    if (count == 10_001)
                    {
                        break;
                    }
                }
            }
            Console.WriteLine(i);
        }
    }
}

File: 007/Problem7.dart
void main() {
  var numLoops = 10;
  final stopwatch = Stopwatch()..start();
  for (var i = 0; i < numLoops; i++) {
    problem();
  }
  final elapsedTime = stopwatch.elapsed;
  print(elapsedTime);
}

bool isPrime(num) {
  for (var i = 2; i < num; i++) {
    if (num % i == 0) {
      return false;
    }
  }
  return true;
}

void problem() {
  var count = 2;
  var i = 4;
  while (true) {
    i++;
    if (isPrime(i)) {
      count++;
      if (count == 10001) {
        break;
      }
    }
  }
  print(i);
}